client/src/components/SaladShow.js

```diff
@@ -1,14 +1,45 @@
 import React, {useState, useEffect } from "react";
 import ReviewList from "./ReviewList";
+import ReviewForm from "./ReviewForm";
+import getNiceDate from "../services/getNiceDate";
+
 
 const SaladShow = (props) =>{
     const [salad, setSalad] = useState({
         name: "",
         description: "",
         reviews: []
     })
-
+    const [errors, setErrors] = useState([])
     const [reviews, setReviews] = useState([])
+
+    const postReview = async (newReview) => {
+        try {
+            const saladId = props.match.params.id
+            const response = await fetch(`/api/v1/salads/${saladId}/reviews`, {
+                method: "POST",
+                headers: new Headers({
+                    "Content-Type": "application/json"
+                }),
+                body: JSON.stringify(newReview)
+            })
+            if (!response.ok) {
+                if(response.status === 422) {
+                    const body = await response.json()
+                    return setErrors(body.errors)
+                } else {
+                    const errorMessage = `${response.status} (${response.statusText})`
+                    const error = new Error(errorMessage)
+                    throw(error)
+                }
+            } else {
+                const body = await response.json()
+                getSalad()
```

Comment:
this is breaking best practices with React by forcing an additional fetch request. The server should be sending back the new Review that can be added to state with the other reviews